frontend/src/components/common/CreateResourceButton.tsx

```diff
@@ -0,0 +1,210 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { getCluster } from '../../lib/cluster';
+import { apply } from '../../lib/k8s/apiProxy';
+import { KubeObjectInterface } from '../../lib/k8s/cluster';
+import { KubeConfigMap } from '../../lib/k8s/configMap';
+import { KubeRuntimeClass } from '../../lib/k8s/runtime';
+import { KubeSecret } from '../../lib/k8s/secret';
+import { clusterAction } from '../../redux/clusterActionSlice';
+import { EventStatus, HeadlampEventType, useEventCallback } from '../../redux/headlampEventSlice';
+import { ActionButton, EditorDialog } from '../common';
+
+const creationTimestamp = new Date('2022-01-01').toISOString();
+const BASE_EMPTY_CONFIG_MAP: KubeConfigMap = {
+  apiVersion: 'v1',
+  kind: 'ConfigMap',
+  metadata: {
+    creationTimestamp: '2023-04-27T20:31:27Z',
+    name: 'my-pvc',
+    namespace: 'default',
+    resourceVersion: '1234',
+    uid: 'abc-1234',
+  },
+  data: {},
+};
+const LEASE_DUMMY_DATA = [
+  {
+    apiVersion: 'coordination.k8s.io/v1',
+    kind: 'Lease',
+    metadata: {
+      name: 'lease',
+      namespace: 'default',
+      creationTimestamp,
+      uid: '123',
+    },
+    spec: {
+      holderIdentity: 'holder',
+      leaseDurationSeconds: 10,
+      leaseTransitions: 1,
+      renewTime: '2021-03-01T00:00:00Z',
+    },
+  },
+];
+const BASE_RC: KubeRuntimeClass = {
+  apiVersion: 'node.k8s.io/v1',
+  kind: 'RuntimeClass',
+  metadata: {
+    name: 'runtime-class',
+    namespace: 'default',
+    creationTimestamp,
+    uid: '123',
+  },
+  handler: 'handler',
+  overhead: {
+    cpu: '100m',
+    memory: '128Mi',
+  },
+  scheduling: {
+    nodeSelector: {
+      key: 'value',
+    },
+    tolerations: [
+      {
+        key: 'key',
+        operator: 'Equal',
+        value: 'value',
+        effect: 'NoSchedule',
+        tolerationSeconds: 10,
+      },
+    ],
+  },
+};
+const BASE_EMPTY_SECRET: KubeSecret = {
+  apiVersion: 'v1',
+  kind: 'Secret',
+  metadata: {
+    creationTimestamp: '2023-04-27T20:31:27Z',
+    name: 'my-pvc',
+    namespace: 'default',
+    resourceVersion: '1234',
+    uid: 'abc-1234',
+  },
+  data: {},
+  type: 'bla',
+};
+
+export interface CreateResourceButtonProps {
+  resource: string;
+}
+
+export function CreateResourceButton(props: CreateResourceButtonProps) {
```

Comment:
AFAIU, the only difference between the CreateButton and this one, is that the this one uses a predefined item. So we are kind of duplicating the apply logic between this button, the CreateButton, and it's also close to the EditorButton.

So, it would be nice to be able to run the creation dialog from any other component. So we could perhaps move the apply logic to the EditorDialog, and then we open it with the code we want, and from where we want.